Team9Project/src/com/indragie/cmput301as1/ExpenseClaimDetailActivity.java

```diff
@@ -225,7 +282,56 @@ public boolean onCreateOptionsMenu(Menu menu) {
 	@Override
 	public boolean onPrepareOptionsMenu(Menu menu) {
 		super.onPrepareOptionsMenu(menu);
-		menu.findItem(R.id.action_add_item).setEnabled(editable);
+		boolean UserCheck = user.getName().contentEquals(claim.getUser().getName());//SHOULD BE ID USING NAME FOR TESTING
+
+		if (status ==Status.APPROVED){
+			menu.findItem(R.id.action_add_item).setEnabled(false);
+			menu.findItem(R.id.action_mark_submitted).setEnabled(false);
+			menu.findItem(R.id.action_mark_approved).setEnabled(false);
+			menu.findItem(R.id.action_mark_returned).setEnabled(false);
+		}
+		if(status == Status.RETURNED){
+			if(UserCheck){
+				menu.findItem(R.id.action_add_item).setEnabled(true);
+				menu.findItem(R.id.action_mark_submitted).setEnabled(true);
+				menu.findItem(R.id.action_mark_approved).setEnabled(false);
+				menu.findItem(R.id.action_mark_returned).setEnabled(false);
```

Comment:
@mrthegreensky Agreed, much of this is repetitive. Suggestions to make it cleaner:

* Get the references to the menu items by calling `findItem()` for each ID once, and store the menu items in temporary variables so you can just call `setEnabled()` on them inside these conditional branches rather than calling `findItem()` every time
* Some of the lines are duplicated inside the conditional statements and could be moved to the parent scope to fix the duplication. See comments below